### ISSUE TITLE
Error creating bridge: Permission denied.

### ISSUE THREAD

**Issue by AdrianPeniak** [eclipse/mosquitto#251]
Hi Guys,
I trying run mosquitto on embedded linux device with configuration (show below) as root user and cmd: mosquitto -c /etc/mosquitto/mosquitto.conf

### /etc/mosquitto/mosquitto.conf ###
log_dest stdout
#mosquitto user
user mosquitto 
#bridge Config
connection 1234567890
address localhost:1883
start_type automatic  
topic # both 0      
topic # both 1
#cleansession false
bridge_protocol_version mqttv311
#need to be false otherwise error on hivemq-side
notifications false 
### end of conf file ###

mosquitto can't created bridge and I catch this log:
1472052730: mosquitto version 1.4.8 (build date 2016-07-05 08:47:49+0200) starting
1472052731: Config loaded from /etc/mosquitto/mosquitto.conf.
1472052731: Opening ipv4 listen socket on port 1883.
1472052731: Opening ipv6 listen socket on port 1883.
1472052731: Connecting bridge 1234567890 (localhost:1883)
1472052731: Error creating bridge: Permission denied.
1472052731: Warning: Unable to connect to bridge 1234567890.
1472052761: Connecting bridge 1234567890 (localhost:1883)

However when I change user in conf file from mosquitto to root (but I know it's not correct) everything works fine. 
Can you help me resolved this problem?

**Comment by ralight** [comment]
I've been looking into this and haven't been able to reproduce it.

I think the only way that this could happen is if there are extra restrictions set up on your system related to creating or binding to sockets.

**Comment by AdrianPeniak** [comment]
Yes you right. This error I received on embedded device. I investigated some more over strace and recieved this log:
write(1, "6730: Connecting bridge <unique "..., 54) = 54
socket(PF_INET, SOCK_STREAM, IPPROTO_TCP) = -1 EACCES (Permission denied)
gettimeofday({6730, 969138}, NULL)      = 0
write(1, "6730: Error creating bridge: Per"..., 48) = 48

Further I found some solution:
- Install libcap and libcap2 on ATM2
- setcap 'cap_net_raw=+ep' /usr/sbin/mosquitto
work fine. But the mosquitto is working when I run mosquitto as mosquitto user only:
- su mosquitto -c “/usr/sbin/mosquitto -c /etc/mosquitto/mosquittoBMW.conf”
thus executing over root still not work:
 /usr/sbin/mosquitto -c /etc/mosquitto/mosquittoBMW.conf

Unfortunately I’m still not 100% sure where was the problem. So how can I resolved this problem at system level (without setcap).

**Comment by ralight** [comment]
Does the problem exist for other utilities that use network sockets?

**Comment by AdrianPeniak** [comment]
Not but everything running as root

**Comment by PierreF** [comment]
Do you still have this issue ? If yes, does your system has some kind of security policyes (SELinux, AppArmor, ...) that may restrict Mosquitto access ?

**Comment by ralight** [comment]
I'm closing this issue because I don't believe it is a specific Mosquitto issue, but related to the system. If you've still got problems and think it is related to Mosquitto, please reopen and add more details.
